model-inference/decisionTree/experiments/test_model.py

```diff
@@ -1,314 +1,292 @@
+            'HummingbirdTorchScriptCPU', 
+            'HummingbirdTVMCPU', 
+            'TFDF',
+            'ONNXCPU', 
+            'HummingbirdPytorchGPU',
+            'HummingbirdTorchScriptGPU',
+            'ONNXGPU'
+        ],
+        help="Framework to run the decision forest model.")
+    parser.add_argument("--batch_size", type=int, 
+        help="Batch size for testing. For Sklearn, TreeLite, ONNX, batch_size will not be used.")
+    parser.add_argument("--query_size", type=int, help="Query size for testing.")
+    args = parser.parse_args()
+    check_argument_conflicts(args)
+    if args.dataset:
+        DATASET = args.dataset
+    if args.model:
+        MODEL = args.model
+    if args.frameworks:
+        FRAMEWORK = args.frameworks
+    if not args.batch_size:
+        args.batch_size = config[DATASET]["batch_size"]
+    if not args.query_size:
+        args.query_size = config[DATASET]["query_size"]
+
+    if DATASET == "year" or DATASET == "airline_regression":
+        args.task_type = "regression"
+    else:
+        args.task_type = "classification"
+    # Print arguments
+    print(f"DATASET: {DATASET}")
+    print(f"MODEL: {MODEL}")
+    print(f"FRAMEWORK: {FRAMEWORK}")
+    print(f"Query Size: {args.query_size}")
+    print(f"Batch Size: {args.batch_size}")
+    return args
+
+
+def load_data(config, time_consume):
+    df_test = fetch_data(DATASET, config, "test", time_consume=time_consume)
+    y_col = config[DATASET]["y_col"]
+    x_col = list(df_test.columns)
+    x_col.remove(y_col)
+    features = df_test[x_col].to_numpy(dtype=np.float32)
+    label = df_test[y_col]
+    return (features, label)
+
+def load_sklearn_model(config, time_consume):
+    start_time = time.time()
+    relative_path = relative2abspath("models", f"{DATASET}_{MODEL}_{config['num_trees']}_{config['depth']}.pkl")
+    sklearnmodel = joblib.load(relative_path)
+    sklearnmodel.set_params(verbose =0)
+    sklearnmodel.set_params(n_jobs =-1)
+    load_time = time.time()
+    sklearnmodel_loading_time = calculate_time(start_time, load_time)
+    time_consume["sklearn loading time"] = sklearnmodel_loading_time
+    print(f"Time Taken to load sklearn model: {sklearnmodel_loading_time}")
+    return sklearnmodel
+
+
+def test(*argv):
+    if FRAMEWORK.endswith("GPU"):
+        test_postprocess(*test_gpu(*argv))
+    else:
+        test_postprocess(*test_cpu(*argv))
+
+def test_cpu(args, features, label, sklearnmodel, config, time_consume):
+    input_size = len(label)
 
     if FRAMEWORK == "Sklearn":
         start_time = time.time()
         #scikit-learn will use all data in a query as one batch  
-        results = run_inference(FRAMEWORK, data, input_size, query_size, sklearnmodel.predict)
-        write_data(FRAMEWORK, results)
-        end_time = time.time()
-        print("TOTAL Time Taken "+FRAMEWORK+" is:", calulate_time(start_time,end_time))
-        find_accuracy(FRAMEWORK,df_test[y_col],results)
-
-        del results
```

Comment:
Were the memory usage, cpu usage, and time-taken, similar when results were not deleted and gc.collect() was not called? Because I remember this taking too much ram and skewing the results when we tested multiple experiments in one go.

Reply:
How did you test multiple experiments in one go?

Reply:
I think this would affect our benchmark results with respect to hardware utilization and time. Maybe, we can discuss and reach a consensus on whether to do this for the experiments before benchmarking.